stacks-coordinator/src/coordinator.rs

```diff
@@ -122,32 +122,67 @@ trait CoordinatorHelpers: Coordinator {
-        self.stacks_node().broadcast_transaction(&tx)?;
-        Ok(())
+        loop {
+            // Build a mint transaction using the peg in op and calculated nonce
+            let mint_tx = self
+                .fee_wallet()
+                .stacks()
+                .build_mint_transaction(&op, nonce)?;
+
+            // Broadcast the resulting sBTC transaction to the stacks node
+            match self.stacks_node().broadcast_transaction(&mint_tx) {
+                Err(StacksNodeError::BroadcastError(BroadcastError::ConflictingNonceInMempool)) => {
+                    warn!("Mint transaction rejected by stacks node due to conflicting nonce in mempool. Stacks node may be falling behind!");
+                    warn!("Incrementing nonce and retrying...");
+                    nonce = self.stacks_node_mut().next_nonce(&address)?;
+                }
+                Err(StacksNodeError::BroadcastError(BroadcastError::FeeTooLow(
+                    expected,
+                    actual,
+                ))) => {
+                    warn!("Mint transaction rejected by stacks node due to provided fee being too low: {}", actual);
+                    warn!("Incrementing fee to {} and retrying...", expected);
+                    self.fee_wallet_mut().stacks_mut().set_fee(expected);
+                }
+                Err(e) => return Err(e.into()),
+                Ok(_) => return Ok(()),
+            }
+        }
     }
 
     fn peg_out(&mut self, op: stacks_node::PegOutRequestOp) -> Result<()> {
         // Retrieve the nonce from the stacks node using the sBTC wallet address
         let address = *self.fee_wallet().stacks().address();
-        let nonce = self.stacks_node_mut().next_nonce(&address)?;
+        let mut nonce = self.stacks_node_mut().next_nonce(&address)?;
 
-        // Build a burn transaction using the peg out request op and calculated nonce
-        let burn_tx = self
-            .fee_wallet()
-            .stacks()
-            .build_burn_transaction(&op, nonce)?;
-
-        // Broadcast the resulting sBTC transaction to the stacks node
-        self.stacks_node().broadcast_transaction(&burn_tx)?;
+        loop {
+            // Build a burn transaction using the withdrawal request op and calculated nonce
+            let burn_tx = self
+                .fee_wallet()
+                .stacks()
+                .build_burn_transaction(&op, nonce)?;
+
+            // Broadcast the resulting sBTC transaction to the stacks node
+            match self.stacks_node().broadcast_transaction(&burn_tx) {
+                Err(StacksNodeError::BroadcastError(BroadcastError::ConflictingNonceInMempool)) => {
+                    warn!("Burn transaction rejected by stacks node due to conflicting nonce in mempool. Stacks node may be falling behind!");
+                    warn!("Incrementing nonce and retrying...");
+                    nonce = self.stacks_node_mut().next_nonce(&address)?;
+                }
```

Comment:
Part 2: Compute the nonce first thing in the loop
```suggestion
        loop {
            // Note: This mutates `self.stacks_node` each iteration
            let nonce = self.stacks_node_mut().next_nonce(&address)?;
            
            // Build a burn transaction using the withdrawal request op and calculated nonce
            let burn_tx = self
                .fee_wallet()
                .stacks()
                .build_burn_transaction(&op, nonce)?;

            // Broadcast the resulting sBTC transaction to the stacks node
            match self.stacks_node().broadcast_transaction(&burn_tx) {
                Err(StacksNodeError::BroadcastError(BroadcastError::ConflictingNonceInMempool)) => {
                    warn!("Burn transaction rejected by stacks node due to conflicting nonce in mempool. Stacks node may be falling behind!");
                    warn!("Incrementing nonce and retrying...");
                }
```

Reply:
I cannot do this. If I did, I would increment the nonce in the fee to low case which is incorrect.